vnext/ReactUWP/Utils/ValueUtils.cpp

```diff
@@ -180,6 +181,31 @@ DateTimeToDynamic(winrt::DateTime dateTime, int64_t timeZoneOffsetInSeconds) {
   return strDateTime;
 }
 
+// In UWP apps, the string provided to FontFamily must be either a system-installed font name or a
+// a special path string to a font file bundled with the app (e.g. "/Assets/MyFont.ttf#MyFont").
+// Since this convention for custom fonts is not portable, the following code provides a mechanism for
+// app developers to register their custom font paths with aliases and use those aliases in React for any
+// FontFamily styles.
+
+static FontManager s_FontManager{};
+
+REACTWINDOWS_API_(void)
+SetFontPath(
+    const std::wstring &fontFamily,
+    winrt::Windows::UI::Text::FontWeight weight,
+    winrt::Windows::UI::Text::FontStyle style,
+    const std::wstring &filePath) {
```

Comment:
>std::wstring  [](start = 10, length = 13)

is this the right type for this? why not std::filesystem::path?

Reply:
A string permits the most flexibility here. The value is fed into the winrt::FontFamily constructor. The value could be just a name, e.g. "Verdana", which allows use of SetFontPath to alias a custom name to a system font. Or it could be URI-like, with an optional protocol, a path to a font file, and a fragment listing the name of the font in the file, e.g. "ms-appx:///Assets/MyFont.ttf#Name", to alias a custom name to a local resource.

---
In reply to: [371631708](https://github.com/microsoft/react-native-windows/pull/3830#discussion_r371631708) [](ancestors = 371631708)